app/Http/Controllers/RegisteredUserController.php

```diff
@@ -6,27 +6,76 @@
 use Illuminate\Support\Facades\Auth;
 use App\Models\User;
 use Illuminate\Validation\Rules\Password;
+use Illuminate\Validation\ValidationException;
 
 class RegisteredUserController extends Controller
 {
     public function create()
     {
-        return view('register');
+        return view('user.create');
     }
 
-    public function store()
+    public function show(Request $request)
     {
-        $validatedAttributes = request()->validate([
-            'name' => ['required'],
+        $isHTMX = $request->hasHeader('HX-Request');
+        return view('user.show', ['isHTMX' => $isHTMX, 'user' => auth()->user()])->fragmentIf($isHTMX, 'show-user');
+    }
+
+    public function store(Request $request)
+    {
+        $data = $request->validate([
+            'name' => ['required', 'unique:users,name'],
             'email' => ['required', 'email', 'unique:users,email'],
             'password' => ['required'],
             'password' => ['required', 'confirmed'],
         ]);
 
-        $user = User::create($validatedAttributes);
+        $user = User::create($data);
 
         Auth::login($user);
 
         return redirect('/');
     }
+
+    public function update(Request $request)
+    {
+        $isHTMX = $request->hasHeader('HX-Request');
+
+        try {
+            $data = $request->validateWithBag('user_errors', [
+                'name' => ['required', \Illuminate\Validation\Rule::unique('users')->ignore(auth()->user()->id)],
```

Comment:
I'm curious why `unique:users` wasn't sufficient here. Was it that the `ignore` wasn't possible?